source/common/http/conn_pool_base.h

```diff
@@ -78,8 +83,13 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
                uint64_t concurrent_stream_limit)
       : Envoy::ConnectionPool::ActiveClient(parent, lifetime_stream_limit,
                                             concurrent_stream_limit) {
-    Upstream::Host::CreateConnectionData data = parent_.host()->createConnection(
-        parent_.dispatcher(), parent_.socketOptions(), parent_.transportSocketOptions());
+    Upstream::Host::CreateConnectionData data =
+        static_cast<Envoy::ConnectionPool::ConnPoolImplBase*>(&parent)->host()->createConnection(
+            parent.dispatcher(), parent.socketOptions(), parent.transportSocketOptions());
+    initialize(data, parent);
+  }
+
+  void initialize(Upstream::Host::CreateConnectionData& data, HttpConnPoolImplBase& parent) {
```

Comment:
Why is this now a separate function? I don't see it called anywhere else. Is this for a future PR?

Reply:
Yeah, there's going to be two constructors in a follow-up PR.
I can move it to the next but if you're OK I'm inclined to leave it.